components/XElement/XElement.astro

```diff
@@ -168,18 +172,43 @@ for (const name in attrs) {
 	}
 }
 
-const onLoadString: string = (
-	listenerString || listenerOfOnce
-		? `(async function($$,$){$=this;$$=(await import($$.src)).default;${
+const Payload: string = `
+	(async function($$,$){$=this;$$=(await import($$.src)).default;${
 			defineVarsString ?? ``
 		}${
 			listenerString ?? ``
 		}${
 			listenerOfOnce ?? ``
-		}}).call(this.previousSibling,this.remove()||this)`
+		}}).call(this.previousSibling,this.remove()||this)
+`
+
+/** Transforms the JS/TS input using ESBuild */
+const transformer = async (code: string): Promise<string> => await transformWithEsbuild(code, 'xelement.ts', { minify: true }).then(result => result.code)
+
+/** Create a hashmap to store the transformed code */
+const hash = Object.create(null)
+
+/** Return transformed code, remembering previously transformed code. */
+const memo = async (code: string) => {
+  //NoOp if the code is undefined, 
+  if (code === undefined) return
+  // if the hashmap has the transformed code for this code, return it
+  if (code in hash) return hash[code]
+
+  // otherwise, the transform code and assign it to the hashmap
+  hash[code] = await transformer(code)
+
+  // and return the transformed code
+  return hash[code]
+}
+
+const onLoadString: string = (
+	listenerString || listenerOfOnce
+		? await memo(Payload)
 	: null
 )
```

Comment:
Suggestion:

```suggestion
const onLoadString: string = (
	listenerString || listenerOfOnce
		? await memo(
			`(async function($$,$){$=this;$$=(await import($$.src)).default;${
				defineVarsString ?? ''
			}${
				listenerString ?? ''
			}${
				listenerOfOnce ?? ''
			}}).call(this.previousSibling,this.remove()||this)`
		)
	: ''
)
```

Then you can delete the `Payload` above (which is confusing, partly because a capital letter usually refers to a constructor, or TypeScript type/interface).

Reply:
this has been accepted